cmd/cluster-controller/main.go

```diff
@@ -75,8 +74,7 @@ func (o *options) Validate() error {
 
 func main() {
 	// Setup signal handler for a cleaner shutdown
-	ctx, cancel := signal.NotifyContext(context.Background(), os.Kill, os.Interrupt)
-	defer cancel()
+	ctx := genericapiserver.SetupSignalContext()
```

Comment:
Re signal handling - sorry - I meant to only do the change specific to syncer. But that's awesome that you've done the others! Would you mind splitting them into a separate PR (and leave the syncer changes in this one)?

Reply:
yeah sure.